libs/estdlib/src/supervisor.erl

```diff
@@ -321,9 +418,16 @@ terminate(_Reason, #state{children = Children} = State) ->
 
 loop_terminate([#child{pid = undefined} | Tail], AccRemaining) ->
     loop_terminate(Tail, AccRemaining);
+loop_terminate([#child{pid = {restarting, _}} | Tail], AccRemaining) ->
+    loop_terminate(Tail, AccRemaining);
 loop_terminate([#child{pid = Pid} = Child | Tail], AccRemaining) when is_pid(Pid) ->
-    do_terminate(Child),
-    loop_terminate(Tail, [Pid | AccRemaining]);
+    case is_process_alive(Pid) of
```

Comment:
I'm not sure about this optimization. The process could be gone between here and the exit message, and it's not worth it unless we need to ensure it's gone.